### PR TITLE
Rewrite tests with mollusk

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR rewrites all program tests with [Mollusk](https://github.com/buffalojoec/mollusk) instead of `solana-program-test`.

### CODE REVIEW

**Review by lorisleiva — APPROVED** [review]

File: program/tests/functional.rs [hunk 1]
```diff
@@ -54,671 +47,494 @@ fn get_config_space(key_len: usize) -> usize {
...
-        (config_keypair.pubkey(), true),
-    ];
+    let signer0 = Pubkey::new_unique();
+    let keys = vec![(pubkey, false), (signer0, true), (config, true)];
     let my_config = MyConfig::new(42);
 
-    create_config_account(
-        &mut context,
-        &config_keypair,
-        vec![(pubkey, false), (pubkey, false), (pubkey, false)], // Dummy keys for account sizing.
```
Comment (joncinque) [inline program/tests/functional.rs:567]: Thank goodness for removing this, it confused the crap out of me during the first review :sweat_smile:

File: program/tests/functional.rs [hunk 2]
```diff
@@ -54,671 +47,494 @@ fn get_config_space(key_len: usize) -> usize {
...
+        &instruction,
+        &[
+            (config, config_account),
+            (signer0, AccountSharedData::default()),
+        ],
+        &[Check::success(), Check::compute_units(3_324)],
+    );
+
+    // Use this for next invoke.
+    let updated_config_account = result.get_account(&config).unwrap().to_owned();
```
Comment (joncinque) [inline program/tests/functional.rs:446]: Is this part actually needed since you're storing the proper account data from the start?

**Review by joncinque — COMMENTED** [review]
Looks really good! Just one little question on whether one of the stores is needed

File: program/tests/functional.rs (hunk 2 above)
Comment (buffalojoec) [reply program/tests/functional.rs:446]: Technically you're right that would also work, but to be extra sure I wanted to pipe the newly created config account state into the "update" instruction. The `create_config_account` helper uses the default value (uninitialized data) to space the account, and the first invoke sets it to `MyConfig::new(42)`. This way we get to assert a valid transition from `MyConfig::new(42)` to `MyConfig::new(84)`.

Comment (joncinque) [reply program/tests/functional.rs]: Ok great, the new change looks good!

**Review by joncinque — APPROVED** [review]

**Pull request merged**